extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js

```diff
@@ -465,9 +501,25 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
   adTimedOut_() {
     return (Date.now() - this.timeCurrentPageCreated_) > TIMEOUT_LIMIT;
   }
+
+
+  /**
+   * @param {string} eventType
+   * @param {!Object<string, string>=} vars A map of vars and their values.
+   * @private
+   */
+  analyticsEvent_(eventType, vars) {
+    const adIndex = this.adPagesCreated_;
+    if (!hasOwn(this.analyticsData_, adIndex.toString())) {
+      this.analyticsData_[adIndex] = {adIndex};
+    }
+    this.analyticsData_[adIndex] = Object.assign(this.analyticsData_[adIndex],
+        vars);
+    triggerAnalyticsEvent(this.element, eventType,
+        this.analyticsData_[adIndex]);
```

Comment:
shall we free up data for previous ads? instead of an object map, we can use a single object here and clean it at new request time.

Reply:
I thought at some point we might want to send data from the previous ads as well. Some sort of batching

Reply:
how are you going to refer to a previous ad? to do that, you will need to turn the macro var to macro function. 

for batching, we should resolve the var at triggering time.

Reply:
changed.